### PR TITLE
[DebugInfo] Add DW_ATE_complex_float translation

### PR DESCRIPTION
**Pull request by MrSidims**
It's mapped on new Encoding value for DebugBasicType in NonSemantic.Shader.DebugInfo.200 spec.
If another DebugInfo instruction set is specified - it's mapped to DW_TAG_unspecified_type

Spec:
https://github.com/KhronosGroup/SPIRV-Registry/pull/186 (TBD)

### CODE REVIEW

**Comment by MrSidims** [comment]
@stanleygambarin please take a look
Appropriate entry to encoding map in the spec will be added a bit later

File: lib/SPIRV/LLVMToSPIRVDbgTran.cpp [hunk 1]
```diff
@@ -572,9 +572,15 @@ SPIRVEntry *LLVMToSPIRVDbgTran::transDbgBaseType(const DIBasicType *BT) {
...
+  // Unset encoding if it's complex and NonSemantic.Shader.DebugInfo.200 is not
+  // enabled
+  if (EncTag == SPIRVDebug::Complex &&
+      BM->getDebugInfoEIS() != SPIRVEIS_NonSemantic_Shader_DebugInfo_200)
+    EncTag = SPIRVDebug::Unspecified;
   Ops[EncodingIdx] = EncTag;
-  if (isNonSemanticDebugInfo())
+  if (isNonSemanticDebugInfo()) {
     transformToConstant(Ops, {EncodingIdx});
+  }
```
Comment (MrSidims) [inline lib/SPIRV/LLVMToSPIRVDbgTran.cpp:583]:
```suggestion
  if (isNonSemanticDebugInfo())
    transformToConstant(Ops, {EncodingIdx});
```

**Comment by MrSidims** [comment]
@vmaksimo @asudarsa @maksimsab @jgstarIntel @LU-JOHN please take a look

File: lib/SPIRV/libSPIRV/SPIRV.debug.h [hunk 2]
```diff
@@ -88,7 +88,8 @@ enum EncodingTag {
   Signed       = 4,
   SignedChar   = 5,
   Unsigned     = 6,
-  UnsignedChar = 7
+  UnsignedChar = 7,
+  Complex = 8
```
Comment (vmaksimo) [inline lib/SPIRV/libSPIRV/SPIRV.debug.h:92]: Please format

File: test/DebugInfo/NonSemantic/Shader200/FortranComplex.ll [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+; Function Attrs: nounwind uwtable
+define void @MAIN__() local_unnamed_addr !dbg !4 !llfort.type_idx !12 {
+alloca_0:
+  %func_result = tail call i32 @for_set_fpe_(ptr nonnull @0), !dbg !13, !llfort.type_idx !14
+  %func_result2 = tail call i32 @for_set_reentrancy(ptr nonnull @1), !dbg !13, !llfort.type_idx !14
+  call void @llvm.dbg.value(metadata float 1.000000e+00, metadata !11, metadata !DIExpression(DW_OP_LLVM_fragment, 0, 32)), !dbg !15
+  call void @llvm.dbg.value(metadata float 2.000000e+00, metadata !11, metadata !DIExpression(DW_OP_LLVM_fragment, 32, 32)), !dbg !15
+  call void @llvm.dbg.value(metadata float 2.000000e+00, metadata !10, metadata !DIExpression(DW_OP_LLVM_fragment, 0, 32)), !dbg !15
+  call void @llvm.dbg.value(metadata float -1.000000e+00, metadata !10, metadata !DIExpression(DW_OP_LLVM_fragment, 32, 32)), !dbg !15
+  call void @llvm.dbg.value(metadata float poison, metadata !8, metadata !DIExpression(DW_OP_LLVM_fragment, 0, 32)), !dbg !15
```
Comment (vmaksimo) [inline test/DebugInfo/NonSemantic/Shader200/FortranComplex.ll:71]: Just to be sure, is it OK that we have debug info for poison value metadata?

File: test/DebugInfo/NonSemantic/Shader200/FortranComplex.ll [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+; CHECK-SPIRV-100-DAG: Constant [[#]] [[#Encoding:]] 0
+; CHECK-SPIRV-100-DAG: ExtInst [[#]] [[#Type:]] [[#Import]] DebugTypeBasic [[#Name]] [[#Size]] [[#Encoding]]
+; CHECK-SPIRV-100-DAG: ExtInst [[#]] [[#]] [[#Import]] DebugLocalVariable [[#]] [[#Type]]
+; CHECK-SPIRV-100-DAG: ExtInst [[#]] [[#]] [[#Import]] DebugLocalVariable [[#]] [[#Type]]
+; CHECK-SPIRV-100-DAG: ExtInst [[#]] [[#]] [[#Import]] DebugLocalVariable [[#]] [[#Type]]
+
+; CHECK-LLVM-100: ![[#]] = !DILocalVariable(name: "a", scope: ![[#]], file: ![[#]], line: [[#]], type: ![[#Type:]])
+; CHECK-LLVM-100: ![[#Type]] = !DIBasicType(tag: DW_TAG_unspecified_type, name: "COMPLEX*8")
+; CHECK-LLVM-100: ![[#]] = !DILocalVariable(name: "b", scope: ![[#]], file: ![[#]], line: [[#]], type: ![[#Type]])
+; CHECK-LLVM-100: ![[#]] = !DILocalVariable(name: "c", scope: !9, file: !3, line: 3, type: ![[#Type]])
```
Comment (vmaksimo) [inline test/DebugInfo/NonSemantic/Shader200/FortranComplex.ll:52]:
```suggestion
; CHECK-LLVM-100: ![[#]] = !DILocalVariable(name: "c", scope: ![[#]], file: ![[#]], line: [[#]], type: ![[#Type]])
```

File: lib/SPIRV/libSPIRV/SPIRV.debug.h (hunk 2 above)
Comment (MrSidims) [reply lib/SPIRV/libSPIRV/SPIRV.debug.h]: Applied

**Review by vmaksimo — APPROVED** [review]

**Pull request merged**